src/lib/badge/badge.spec.ts

```diff
@@ -0,0 +1,93 @@
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {Component, DebugElement} from '@angular/core';
+import {By} from '@angular/platform-browser';
+import {MatBadge, MatBadgeModule} from './index';
+import {ThemePalette} from '@angular/material/core';
+
+describe('MatBadge', () => {
+  let fixture: ComponentFixture<any>;
+  let testComponent: BadgeWithTextContent;
+  let badgeNativeElement;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [MatBadgeModule],
+      declarations: [BadgeWithTextContent],
+    });
+
+    TestBed.compileComponents();
```

Comment:
`compileComponents` completes asynchronously, so it should be in a separate `async` beforeEach preceding the rest of the setup. It only works coincidentally now because the component currently has no `templateUrl` or `styleUrls`, but it would start failing if that ever changes.